core/src/main/java/com/amplitude/core/utilities/InMemoryResponseHandler.kt

```diff
@@ -76,9 +76,13 @@ internal class InMemoryResponseHandler(
             )
             return
         }
-        eventPipeline.flushSizeDivider.incrementAndGet()
-        eventsList.forEach {
-            eventPipeline.put(it)
+
+        scope.launch(storageDispatcher) {
+            eventPipeline.flushSizeDivider.incrementAndGet()
+            eventsList.forEach {
+                delay(BACK_OFF)
+                eventPipeline.put(it)
+            }
```

Comment:
I don't like to add this, but the current response handler integration tests are made in such a way that requires adding delays to control the flow of the test. This requires a lot more effort to fix, so it might be good to revisit when we're dealing with `InMemoryStorage` tasks.